src/main/java/de/maxhenkel/pipez/blocks/PipeBlock.java

```diff
@@ -295,7 +327,14 @@ public boolean isAbleToConnect(LevelAccessor world, BlockPos pos, Direction faci
         return isPipe(world, pos, facing) || canConnectTo(world, pos, facing);
     }
 
-    public abstract boolean canConnectTo(LevelAccessor world, BlockPos pos, Direction facing);
+    public boolean canConnectTo(LevelAccessor world, BlockPos pos, Direction facing) {
+        BlockEntity te = world.getBlockEntity(pos.relative(facing));
```

Comment:
`world.getBlockEntity` is already being called in the implemetations of this method, thus this should have its own method (see comment in EnergyPipeBlock)